libff/algebra/fields/binary/utils.hpp

```diff
@@ -0,0 +1,117 @@
+/**@file
+ *****************************************************************************
+ API hack for treating fields as additive or multiplicative
+ *****************************************************************************
+ * @author     This file is part of libff (see AUTHORS), migrated from libiop
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBFF_ALGEBRA_FIELDS_UTILS_HPP
+#define LIBFF_ALGEBRA_FIELDS_UTILS_HPP
+
+#include "libff/algebra/fields/binary/gf64.hpp"
+#include "libff/algebra/fields/binary/gf128.hpp"
+#include "libff/algebra/fields/binary/gf192.hpp"
+#include "libff/algebra/fields/binary/gf256.hpp"
+
+#include "libff/algebra/fields/prime_base/fp.hpp"
+
+namespace libff {
+
+template<typename FieldT>
+struct is_additive {
+    static const bool value = false;
+};
+
+template<>
+struct is_additive<gf64> {
+    static const bool value = true;
+};
+
+template<>
+struct is_additive<gf128> {
+    static const bool value = true;
+};
+
+template<>
+struct is_additive<gf192> {
+    static const bool value = true;
+};
+
+template<>
+struct is_additive<gf256> {
+    static const bool value = true;
+};
+
+template<typename FieldT>
+struct is_multiplicative {
+    static const bool value = false;
+};
+
+template<mp_size_t n, const bigint<n>& modulus>
+struct is_multiplicative<Fp_model<n, modulus>> {
+    static const bool value = true;
+};
+
+enum field_type {
+    multiplicative_field_type = 1,
+    additive_field_type = 2
+};
```

Comment:
Probably a bad name from libiop (my fault). That it is what it means. (Or in other words,  is the 'smooth' part of the field in the multiplicative subgroup, or in a subspace of the Field)

Reply:
I see, if not already mentioned in the code (I may have missed it), it may be worth dropping a line about that (basically your comment above), so that further confusions are avoided.

Reply:
Yeah, I never touched this part of the code

Reply:
Added the comment